### PR TITLE
Check volumespec is nil in FindPluginBySpec

### PR DESCRIPTION
**Pull request by jingxu97**
This PR fixes issue #49513

### CODE REVIEW

**Comment by msau42** [comment]
Can you add a unit test for this?

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -387,6 +387,10 @@ func (pm *VolumePluginMgr) FindPluginBySpec(spec *Spec) (VolumePlugin, error) {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
+	if spec == nil {
+		return nil, fmt.Error("Could not find plugin because volume spec is nil")
```
Comment (rootfs) [inline pkg/volume/plugins.go]: Errorf

Comment (jingxu97) [inline pkg/volume/plugins.go]: fixed

**Comment by jingxu97** [comment]
@msau42 Added the unit test, PTAL

File: pkg/volume/plugins.go [hunk 2]
```diff
@@ -387,6 +387,10 @@ func (pm *VolumePluginMgr) FindPluginBySpec(spec *Spec) (VolumePlugin, error) {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
+	if spec == nil {
+		return nil, fmt.Errorf("Could not find plugin because volume spec is nil")
+	}
+
```
Comment (gnufied) [inline pkg/volume/plugins.go]: Do we want to remove various checks that were added to guard for nil spec, because I suppose those would be caught by this.

Comment (jingxu97) [inline pkg/volume/plugins.go]: I think we could do that with a separate PR.
